client/components/pagination/index.js

```diff
@@ -84,7 +54,16 @@ class Pagination extends Component {
 	}
 
 	onPageValueChange( event ) {
-		this.setState( { pageInputValue: event.target.value } );
+		const { onPageChange } = this.props;
+		const newPage = parseInt( event.target.value, 10 );
+
+		if ( isFinite( newPage ) ) {
```

Comment:
I think we still want to compare against `this.pageCount` before updating the URL -- otherwise this can happen:

<img width="722" alt="screen shot 2018-08-01 at 8 27 08 am" src="https://user-images.githubusercontent.com/689165/43521644-c670d730-9564-11e8-9647-900f78e7820c.png">